src/commands/runtime/action/create.js

```diff
@@ -182,18 +186,32 @@ class ActionCreate extends RuntimeBaseCommand {
       if (limits) options.limits = limits
       if (paramsAction) options.parameters = paramsAction
       if (annotationParams) options.annotations = annotationParams
-
       const ow = await this.wsk()
-      const method = this.isUpdate() ? 'update' : 'create'
-      const result = await ow.actions[method]({ name, action: options })
-      if (flags.json) {
-        this.logJSON('', result)
-      }
+      await this.syncAction(ow, name, options, flags, method, super.logJSON.bind(this))
     } catch (err) {
-      const method = this.isUpdate() ? 'update' : 'create'
       this.handleError(`failed to ${method} the action`, err)
     }
   }
+
+  /** @private
+   * @param {OpenwhiskClient} ow
+   * @param {string} actionName
+   * @param {object} actionData
+   * @param {'update' | 'create'} method
+   * @param {object} flags
+   * @param {function} logger
+   *
+   * */
+  async syncAction (ow, actionName, actionData, flags, method, logger) {
+    let action = actionData
+    if (flags.copy) {
+      action = await ow.actions.get(flags.copy)
```

Comment:
should this be ow.actions.get(actionName) ?? flags.copy is a boolean

Reply:
Flags.copy is the string value of the action you want to copy.
actionName is the name of the new action.